src/pj_obs_api.c

```diff
@@ -222,41 +266,150 @@ PJ_COORD proj_trans_coord (PJ *P, enum proj_direction direction, PJ_COORD coo) {
 }
 
 
-/* Measure numerical deviation after n roundtrips fwd-inv (or inv-fwd) */
-double proj_roundtrip (PJ *P, enum proj_direction direction, int n, PJ_OBS obs) {
-    int i;
-    PJ_OBS o, u;
 
+/*************************************************************************************/
+int proj_transform (
+    PJ *P,
+    enum proj_direction direction,
+    size_t stride,
+    double *x, size_t nx,
+    double *y, size_t ny,
+    double *z, size_t nz,
+    double *t, size_t nt
+) {
+/**************************************************************************************
+
+    Transform a series of coordinates, where the individual coordinate dimension
+    may be represented by an array that is either
+
+        1. fully populated
+        2. a null pointer and/or a length of zero, which will be treated as a
+           fully populated array of zeroes
+        3. of length one, i.e. a constant, which will be treated as a fully
+           populated array of that constant value
+
+    The stride represents the step length, in bytes, between consecutive elements
+    of the array, the idea being that proj_transform can handle transformation of
+    a large class of application specific data structures, without necessarily
+    understanding the structure format, as in:
+
+        typedef struct {double x, y; int quality_level; char surveyor_name[134];} XYQS;
+        XYQS survey[345];
+        double height = 23.45;
+        PJ *P = {...};
+        ...
+        proj_transform (
+            P, PJ_INV, sizeof(XYQS),
+            &(survey[0].x), 345,     (*  We have 345 eastings  *)
+            &(survey[0].y), 345,     (*  ...and 345 northings. *)
+            &height, 1,              (*  The height is the constant  23.45 m *)
+            0, 0                     (*  and the time is the constant 0.00 s *)
+        );
+
+    This is similar to the inner workings of the pj_transform function, but the
+    idea of the stride has been generalized to work for any size of basic unit,
+    not just a number of doubles.
+
```

Comment:
Sure - there is a lot of proper documentation writing ahead - right now I'm just trying to get the most important things in as comments. I will add the thing about returning the number of coordinates actually transformed.